Add how to encrypt/decrypt image and volume (#636)
harvester/docs: docs/volume/volume-security.md

Since Harvesters uses Longhorn to encrypt volume, we need to pass required fields to Longhorn CSI driver.

![](/img/v1.4/image/create-storage-class.png)

Example Storage Class:

```yaml
allowVolumeExpansion: true
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: encryption
parameters:
  csi.storage.k8s.io/node-publish-secret-name: encryption
  csi.storage.k8s.io/node-publish-secret-namespace: default
  csi.storage.k8s.io/node-stage-secret-name: encryption
  csi.storage.k8s.io/node-stage-secret-namespace: default
  csi.storage.k8s.io/provisioner-secret-name: encryption
  csi.storage.k8s.io/provisioner-secret-namespace: default
  encrypted: "true"
  migratable: "true"
  numberOfReplicas: "3"
  staleReplicaTimeout: "2880"
provisioner: driver.longhorn.io
reclaimPolicy: Delete
volumeBindingMode: Immediate
```

:::info important

If you'd like to select secret in the system namespace. You can use Edit as YAML or kubectl to create the secret. On the Harvester Secrets page, Harvester does not display resources in the system namespace.

:::

## Encrypt Volume

There are two ways to create an encrypted volume. In either case, be sure to select an encrypted storage class.

> ### In Volumes Tabs

Reviewer: Maybe "Volumes page"?

> ![create-empty-volume](/img/v1.2/volume/create-empty-volume.png)

Reviewer: would be nice to update the two screenshots with the encrypted storage class created in the previous steps.

> ### In Images Tab

Reviewer: And "Volumes tab when creating a virtual machine".

> ## Encrypt Volume
> 
> There are two ways to create an encrypted volume. In either case, be sure to select an encrypted storage class.
> 
> ### Volumes Page
> 
> ![create-empty-volume](/img/v1.4/volume/create-empty-volume.png)
> 
> ### Volumes tab when creating a virtual machine
> 
> When creating a virtual machine image, you can add a volume by clicking the `Add Volume button` in the Volumes tab on the virtual machine image page.
> 
> ![create-empty-volume](/img/v1.4/volume/create-empty-volume-in-vm.png)

Reviewer 2: ```suggestion
## Encrypted Volumes

You can create an encrypted volume using an encrypted StorageClass on the following UI locations:

- **Volumes: Create** screen

    ![create-empty-volume](/img/v1.4/volume/create-empty-volume.png)

- **Volumes** tab on the **Virtual Machine: Create** screen

    ![create-empty-volume](/img/v1.4/volume/create-empty-volume-in-vm.png)
```

> ## Advanced usage with Rancher Integration
> 
> ### Prevent other users from reading the secret
> 
> Since the secret is a base64 encoded string, it's not really encrypted. So, admin might want keep this secret safe. With Rancher Integration, we can use project and namespace to isolate permission. Please check [Multi-Tenancy](../rancher/virtualization-management#multi-tenancy) for more detail.

Reviewer 2: ```suggestion
## Advanced Usage with Rancher Integration

The secret is an unencrypted Base64-encoded string. To keep the secret safe, you can use projects and namespaces to isolate permissions. For more information, see [Multi-Tenancy](../rancher/virtualization-management#multi-tenancy).
```

> ## Limitations
> 
> - Don't support that export to image from encrypted volumes.
> - Don't support that data volume restoring from encrypted to unencrypted or unencrypted to encrypted one.

Reviewer 2: ```suggestion
## Limitations

You cannot perform the following actions:

- Export a volume from an encrypted image to a new image
- Restore an encrypted volume to an unencrypted volume
- Restore an unencrypted volume to an encrypted volume
```
